android/app/src/main/java/au/com/robin/sms/MainActivity.kt

```diff
@@ -0,0 +1,67 @@
+package au.com.robin.sms
+
+import android.Manifest
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import android.widget.Button
+import android.widget.EditText
+import android.widget.Toast
+import androidx.annotation.RequiresPermission
+import androidx.appcompat.app.AppCompatActivity
+
+class MainActivity : AppCompatActivity() {
+
+    /**
+     * Constant variables
+     */
+    companion object {
```

Comment:
What's the difference between:
```kt
class MainActivity:
  const val SIM_SLOT = 
```
VS
```kt
class MainActivity:
  companion object {
    conts val SIM_SLOT =
  }
```

Reply:
This is because from the Kotlin itself. It raises an error `Const 'val' are only allowed on top level, in named objects, or in companion objects`. This is also due to companion object that can be accessed without an instance of the class. If we were to have this 
```kotlin
class MainActivity:
  const val SIM_SLOT = 
```
then we need to have the instance of MainActivity as well, hence it's not allowed. 

A discussion about this between experts can be found here:
- [Allow `const val` in classes and interface](https://youtrack.jetbrains.com/issue/KT-16900)

Reply:
I'd rather have it outside the class then. Anyway it's a minor thing.

Reply:
Okay, got it 👍